integration-tests/utils/integration-helpers.js

```diff
@@ -231,24 +305,187 @@ const deleteRuntimes = async ({ page, billingProject, workspaceName }) => {
+            return 'Deleted';
+          }
+          throw response;
+        }
+      },
+      workspaceId,
+      appName
+    );
+  } while (newStatus === 'Deleting' && count < 18);
+
+  if (newStatus === 'Deleted') {
+    console.log(`deleted app: ${appName}`);
+    return true;
+  }
+  console.error(`delete app ${appName} failed: now in ${newStatus}`);
+  return false;
+};
+
+/**
+ * Delete a v2 disk, returning `true` when deletion is complete.
+ * Will return `false` if disk is not deletable.
+ */
+const fullyDeleteDisk = async (page, disk) => {
+  if (!disk) {
+    console.log('Disk does not exist');
+    return true;
+  }
+  const { id, cloudContext, name, status } = disk;
+  const isDeletable = isResourceDeletable('disk', status);
+  if (!isDeletable) {
+    console.error(`Cannot delete disk ${name} in workspace ${cloudContext} with status ${status}. Try deleting it manually.`);
+    return false;
+  }
+
+  await page.evaluate(async (diskId) => {
+    const disksV2Api = window.Ajax().Disks.disksV2();
+    await disksV2Api.delete(diskId);
+  }, id);
+
+  let newStatus = status;
+  let count = 0;
+  do {
+    count++;
+    await delay(Millis.ofSeconds(10));
+    newStatus = await page.evaluate(
+      async (cloudContext, diskName) => {
+        try {
+          const diskV1Api = window.Ajax().Disks.disksV1().disk(cloudContext, diskName);
+          const diskState = await diskV1Api.details();
+          return diskState.status;
+        } catch (response) {
+          if (response.status === 404) {
+            return 'Deleted';
+          }
+          throw response;
+        }
+      },
+      cloudContext,
+      name
+    );
+  } while (newStatus === 'Deleting' && count < 18);
+  if (newStatus === 'Deleted') {
+    console.log(`deleted disk: ${name}`);
+    return true;
+  }
+  console.error(`delete disk ${name} failed: now in ${newStatus}`);
+  return false;
+};
+
+/**
+ * Delete a v2 runtime, returning `true` when deletion is complete. Deletes the associated disk.
+ * Will return `false` if runtime or disk is not deletable.
+ */
+const fullyDeleteRuntime = async (page, runtime) => {
+  const { workspaceId, runtimeName, status } = runtime;
+  // TODO [IA-4337] fix disk delete as part of runtime delete
+  const isDeletable = isResourceDeletable('runtime', status);
+  if (!isDeletable) {
+    console.error(`Cannot delete runtime ${runtimeName} in workspace ${workspaceId} with status ${status}. Try deleting it manually.`);
+    return false;
+  }
+
+  await page.evaluate(
+    async (workspaceId, runtimeName) => {
+      const runtimeApi = window.Ajax().Runtimes.runtimeV2(workspaceId, runtimeName);
+      const willDeleteDisk = true;
+      await runtimeApi.delete(willDeleteDisk);
+    },
+    workspaceId,
+    runtimeName
+  );
+
+  let newStatus = status;
+  let count = 0;
+  do {
+    count++;
+    await delay(Millis.ofSeconds(10));
+    newStatus = await page.evaluate(
+      async (workspaceId, runtimeName) => {
```

Comment:
Isn't this code also above at line 360?

Reply:
I guess there are actually 3 different versions that have a lot of the same boilerplate wrapping. It would be nice if the duplication could be removed.

Reply:
It's very similar to the deleteApp code; both call delete and then poll until delete is complete.